File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.py
def solution(S):
    p = [x for x in S if (x == 'p') | (x == 'P')]
    y = [x for x in S if (x == 'y') | (x == 'Y')]
    if len(p) == len(y):
        return True
    elif (len(p) == 0) & (len(y) == 0):
        return True
    else:
        return False

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var ans:Bool = false
    var pCount : Int = 0
    var yCount : Int = 0
    for countString in s {
        if (countString == "p") || (countString == "P") {
            pCount += 1
        }
        if (countString == "y") || (countString == "Y") {
            yCount += 1
        }
    }
    if (pCount == 0) && (yCount == 0) {
        ans = true
    }else if (pCount == yCount) {
        ans = true
    }else if (pCount != yCount) {
        ans = false
    }
    return ans
}